lightning/src/routing/network_graph.rs

```diff
@@ -506,6 +540,56 @@ impl NetworkGraph {
 		None
 	}
 
+	/// Increments payment_sent_score for appropriate DirectionalChannelInfos on the network graph.
+	/// The appropriate directional channel is indentified by the RouteHop, which includes a
+	/// PublicKey field. This is matched against the two PublicKey fields in ChannelInfo and is
+	/// used to increment the score.
+	pub fn score_payment_sent_for_route(&mut self, route: Vec<Vec<RouteHop>>) -> Result<bool, LightningError> {
+		for path in route {
+			for route_hop in path {
+				let short_channel_id = route_hop.short_channel_id;
+				let channel = self.channels.get_mut(&short_channel_id).unwrap();
+				let directional_channel_info = if route_hop.pubkey == channel.node_one {
+					channel.one_to_two.as_mut().unwrap()
+				} else {
+					channel.two_to_one.as_mut().unwrap()
+				};
+				let channel_score = &mut directional_channel_info.channel_score;
+				channel_score.payment_sent_score += 1;
+			}
+		}
+		Ok(true)
+	}
+
+	/// Increments payment_failed_score for approriate DirectionalChannelInfos on the network
+	/// graph. Like the case for scoring PaymentSent events, this method uses the node's PublicKey
+	/// to identify the appropriate DirectionalChannelInfo to score. From there, there is a check
+	/// against a list of at fault nodes that determines whether the node's channel score should be
+	/// penalized for failing to execute or rewarded for executing properly.
+	pub fn score_payment_failed_for_route(&mut self, route: Vec<Vec<RouteHop>>, faultive_nodes: Vec<PublicKey>) -> Result<bool, LightningError> {
+		if faultive_nodes.len() == 0 {
```

Comment:
I think this case is an API misuage, you shouldn't call `score_payment_failed` if there is no `faultive_nodes` ?

Reply:
I agree. I have removed the if{}.